app/models/product_drive.rb

```diff
@@ -76,15 +77,29 @@ def item_quantities_by_name_and_date(date_range)
     end
   end
 
-  def donation_quantity_by_date(date_range)
-    donations.joins(:line_items)
-      .during(date_range)
-      .sum('line_items.quantity')
+  def donation_quantity_by_date(date_range, item_category_id = nil)
+    if [nil, ''].include? item_category_id
+      donations.joins(:line_items)
+        .during(date_range)
+        .sum('line_items.quantity')
+    else
+      donations.joins(line_items: [:item])
+        .where(item: {item_category_id: item_category_id})
+        .during(date_range)
+        .sum('line_items.quantity')
+    end
   end
 
-  def distinct_items_count_by_date(date_range)
-    donations.joins(:line_items)
-      .during(date_range)
-      .select('line_items.item_id').distinct.count
+  def distinct_items_count_by_date(date_range, item_category_id = nil)
+    if [nil, ''].include? item_category_id
+      donations.joins(:line_items)
+        .during(date_range)
+        .select('line_items.item_id').distinct.count
+    else
+      donations.joins(line_items: [:item])
+        .where(item: {item_category_id: item_category_id})
+        .during(date_range)
+        .select('line_items.item_id').distinct.count
```

Comment:
Same as above.

Reply:
See above comment, can this be built reusing the parts that are common to both conditions?